src/main/java/org/lightcouch/internal/URIBuilder.java

```diff
@@ -54,17 +54,33 @@ public static URIBuilder buildUri(URI uri) {
 
 	public URI build() {
 		try {
+			StringBuilder queryBuilder = new StringBuilder(encodeQuery(query.toString()));
+
 			for (int i = 0; i < qParams.size(); i++) {
 				String amp = (i != qParams.size() - 1) ? "&" : "";
-				query.append(qParams.get(i) + amp);
+				String param;
+				String queryParam = qParams.get(i);
+				int firstEqualsIndex = queryParam.indexOf('=');
```

Comment:
Addressed in aa78892 & 136cc50:
* aa78892 deprecates the methods that don't keep keys and values separate from each other;
* 136cc50 stores the query parameters internally as key/value pairs. `Params` now stores a list of `Param` rather than `String`, where a `Param` is a key/value pair.  The `Param` class has a method `toURLEncodedString` which handles the URL encoding of the query parameters.